packages/react-devtools-scheduling-profiler/src/context/Contexts.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {createContext} from 'react';
+
+export type ShowFn = ({|data: Object, pageX: number, pageY: number|}) => void;
+export type HideFn = () => void;
+export type OnChangeFn = boolean => void;
+
+const idToShowFnMap = new Map<string, ShowFn>();
+const idToHideFnMap = new Map<string, HideFn>();
+
+let currentHideFn: ?HideFn = null;
+let currentOnChange: ?OnChangeFn = null;
+
+function hideMenu() {
+  if (typeof currentHideFn === 'function') {
+    currentHideFn();
+
+    if (typeof currentOnChange === 'function') {
+      currentOnChange(false);
+    }
+  }
+
+  currentHideFn = null;
+  currentOnChange = null;
+}
+
+function showMenu({
+  data,
+  id,
+  onChange,
+  pageX,
+  pageY,
+}: {|
+  data: Object,
+  id: string,
+  onChange?: OnChangeFn,
+  pageX: number,
+  pageY: number,
+|}) {
```

Comment:
Ah, nice catch! I'll do that.

Should I also make this change in `react-devtools-shared`'s implementation too? Also, would it be a good idea to merge the 2 context menu implementations after this PR? I think the only differences are that the scheduling profiler's implementation has an extra `onChange` callback (from your prototype) and slightly more comprehensive Flow types (which I added), as well as some CSS changes.

Reply:
Yeah if the same bug exists in `react-devtools-shared'` it would be nice to do a fix for that as well (maybe as a separate PR tho :grin:)

Let's not do any code-sharing yet.